wagtail/admin/forms/account.py

```diff
@@ -22,10 +22,17 @@
 class NotificationPreferencesForm(forms.ModelForm):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        user_perms = UserPagePermissionsProxy(self.instance.user)
-        if not user_perms.can_publish_pages():
+
+        permission_policy = PagePermissionPolicy()
+        publishable_pages = permission_policy.instances_user_has_permission_for(
+            self.instance.user, "publish"
+        )
+        editable_pages = permission_policy.instances_user_has_permission_for(
+            self.instance.user, "edit"
```

Comment:
While I'm satisfied that this is preserving the previous behaviour of `can_publish_pages` / `can_edit_pages`, I see no reason why these couldn't just be `permission_policy.user_has_permission(self.instance.user, "edit")`. I think the only functional difference is that it would return false for users with 'add' permission who haven't created any pages yet - and it seems perverse to not let them set their notification preferences until they _have_ done.